### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by Stanislava88**
Here i have solved the task for send a message through an external system using TDD and Jmock

### CODE REVIEW

File: src/com/clouway/message/Message.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class Message {
+    public final String recipient;
+    public final String title;
+    public final String content;
+
+    public Message(String recipient, String title, String content) {
+
```
Comment (freekman) [inline src/com/clouway/message/Message.java]: remove line

File: tests/com/clouway/message/MessageDispatcherTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class MessageDispatcherTest {
+    @Rule
+    public JUnitRuleMockery context = new JUnitRuleMockery();
+    private MessageGateway gateway = context.mock(MessageGateway.class);
+    private Validator validator = context.mock(Validator.class);
+
+    MessageDispatcher dispatcher = new MessageDispatcher(gateway, validator);
```
Comment (freekman) [inline tests/com/clouway/message/MessageDispatcherTest.java]: private

File: tests/com/clouway/message/MessageDispatcherTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class MessageDispatcherTest {
+    @Rule
+    public JUnitRuleMockery context = new JUnitRuleMockery();
+    private MessageGateway gateway = context.mock(MessageGateway.class);
```
Comment (freekman) [inline tests/com/clouway/message/MessageDispatcherTest.java]: i have told you to separate the logically joined code

File: tests/com/clouway/message/MessageDispatcherTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            will(returnValue(true));
+
+            oneOf(gateway).send(message);
+        }});
+
+        dispatcher.send(message);
+    }
+
+    @Test
+    public void sendMessageWithoutTitle() throws Exception {
```
Comment (freekman) [inline tests/com/clouway/message/MessageDispatcherTest.java]: what are you really testing here:
- send without title ?

and if so what about send without recipient, send without contend and so on ......................
or are you really testing that the message will not be send

File: src/com/clouway/message/MessageRangeValidator.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package com.clouway.message;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class MessageRangeValidator implements Validator {
```
Comment (freekman) [inline src/com/clouway/message/MessageRangeValidator.java]: OK  ```MessageRangeValidator```,```MessageValidator``` if you need another message validator then what?
I think you should have only one Message validator to deal  with the validation of a message

File: tests/com/clouway/message/MessageDispatcherTest.java (hunk 4 above)
Comment (Stanislava88) [inline tests/com/clouway/message/MessageDispatcherTest.java]: here i test that message not be send without title

File: tests/com/clouway/message/MessageRangeValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    public void happyPath() throws Exception {
+        Message message = new Message("recipient", "title", "content");
+        boolean result = validator.isValid(message);
+
+        assertThat(result, is(true));
+    }
+
+    @Test
+    public void detectedIsMessageOutOfRange() throws Exception {
+        Message message = new Message("recipient", "title", "content" +
```
Comment (freekman) [inline tests/com/clouway/message/MessageRangeValidatorTest.java]: find a better way to represent repeating Strings

File: tests/com/clouway/message/MessageValidatorTest.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        boolean result = validator.isValid(message);
+
+        assertThat(result, is(true));
+    }
+
+    @Test
+    public void detectIsMessageWithoutRecipient() throws Exception {
+        Message message = new Message("", "title", "content");
+
+        try {
```
Comment (freekman) [inline tests/com/clouway/message/MessageValidatorTest.java]: it is a bad practice to have try{}catch{} in a test so in this case they should be removed

**Comment by Stanislava88** [comment]
done

**Comment by freekman** [comment]
Ass i can see you really like exceptions
take a look at this 
https://dzone.com/articles/how-to-properly-use-exceptions-in-java

**Comment by Stanislava88** [comment]
Exceptions are removed.  Repeating Strings is represent using method repeat()

**Comment by Stanislava88** [comment]
done

File: tests/com/clouway/message/MessageRangeValidatorTest.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class MessageRangeValidatorTest {
```
Comment (freekman) [inline tests/com/clouway/message/MessageRangeValidatorTest.java]: i think you don't need this class

**Comment by freekman** [comment]
Is this PL ready for review

**Comment by Stanislava88** [comment]
Yes

**Comment by freekman** [comment]
LGTM

**Pull request merged**